renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -51,6 +51,77 @@ VkDeviceSize IndexTypeSize(VkIndexType type)
 }
 }
 
+DECLARE_STRINGISE_TYPE(VkAccelerationStructureInfo::GeometryData::Triangles);
+DECLARE_STRINGISE_TYPE(VkAccelerationStructureInfo::GeometryData::Aabbs);
+DECLARE_STRINGISE_TYPE(VkAccelerationStructureInfo::GeometryData);
+DECLARE_STRINGISE_TYPE(VkAccelerationStructureInfo);
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, VkAccelerationStructureInfo::GeometryData::Triangles &el)
+{
+  SERIALISE_MEMBER(vertexFormat);
+  SERIALISE_MEMBER(vertexStride);
+  SERIALISE_MEMBER(maxVertex);
+  SERIALISE_MEMBER(indexType);
+  SERIALISE_MEMBER(hasTransformData);
+}
+INSTANTIATE_SERIALISE_TYPE(VkAccelerationStructureInfo::GeometryData::Triangles);
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, VkAccelerationStructureInfo::GeometryData::Aabbs &el)
+{
+  SERIALISE_MEMBER(stride);
+}
+INSTANTIATE_SERIALISE_TYPE(VkAccelerationStructureInfo::GeometryData::Aabbs);
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, VkAccelerationStructureInfo::GeometryData &el)
+{
+  SERIALISE_MEMBER(geometryType);
+  SERIALISE_MEMBER_TYPED(VkGeometryFlagBitsKHR, flags).TypedAs("VkGeometryFlagsKHR"_lit);
+  SERIALISE_MEMBER(memSize);
+
+  SERIALISE_MEMBER(tris);
+  SERIALISE_MEMBER(aabbs);
+
+  SERIALISE_MEMBER(buildRangeInfo);
+}
+INSTANTIATE_SERIALISE_TYPE(VkAccelerationStructureInfo::GeometryData);
+
+template <typename SerialiserType>
+void DoSerialise(SerialiserType &ser, VkAccelerationStructureInfo &el)
+{
+  SERIALISE_MEMBER(type);
+  SERIALISE_MEMBER_TYPED(VkBuildAccelerationStructureFlagBitsKHR, flags)
+      .TypedAs("VkBuildAccelerationStructureFlagsKHR"_lit);
+  SERIALISE_MEMBER(geometryData);
+}
+INSTANTIATE_SERIALISE_TYPE(VkAccelerationStructureInfo);
+
+uint64_t VkAccelerationStructureInfo::GeometryData::GetSerialisedSize() const
+{
+  const uint64_t aabbsSize = sizeof(VkDeviceSize);    // stride
+
+  // You can't just use sizeof(Triangles) due to padding
```

Comment:
I think you could use `sizeof()` here to reduce the calculation for this? The size returned by `GetSize_InitialState` only has to be an upper bound so we can write the chunk header as we go without rewinding to fixup. As long as it's not a hugely pessimistic upper bound a little bit of padding per chunk is fine and will be dominated by the data itself.